### PR TITLE
connect closures by Yevhen Sahach

### PR DESCRIPTION
**Pull request by eugeneSahach**

### CODE REVIEW

File: src/connect.js [hunk 1]
```diff
@@ -20,7 +20,11 @@
  * @return {Function}
  */
 function connect(params) {
-  // write code here
+  const baseParams = params;
```
Comment (danheim) [inline src/connect.js]: не нужно

File: src/connect.js [hunk 2]
```diff
@@ -20,7 +20,11 @@
...
+
+  return function device(extraParams) {
+    return Object.assign({}, baseParams, extraParams);
```
Comment (danheim) [inline src/connect.js]: давай через оператор spread

**Review by danheim — CHANGES_REQUESTED** [review]

**Review by irenhh — CHANGES_REQUESTED** [review]
what about other comments?

File: src/connect.js [hunk 3]
```diff
@@ -19,8 +19,10 @@
  * @param {Object} baseParams
  * @return {Function}
  */
-function connect(params) {
-  // write code here
+function connect(...baseParams) {
+  return function device(...extraParams) {
+    return Object.assign({}, ...baseParams, ...extraParams);
```
Comment (irenhh) [inline src/connect.js]: `return { ...baseParams, ...extraParams}`

**Review by irenhh — CHANGES_REQUESTED** [review]

**Review by irenhh — APPROVED** [review]

**Pull request closed without merging**